### PR TITLE
Refactor `get_configuration`

### LINKED ISSUE #182: Refactor `get_configuration` into class-based exporter using `ConfigFile`
## Background
`get_configuration` in `src/dfastbe/gui/configs.py` builds a `ConfigParser` directly from GUI state. The function is long and mixes data extraction, section assembly, and conditional logic. We want to refactor it into a class with single-responsibility methods and consider using `ConfigFile` to centralize config writing rules.

## Goals
- Replace the monolithic `get_configuration` function with a class-based exporter.
- Keep output identical to the current `ConfigParser` for version `1.0`.
- Use `ConfigFile` where it improves clarity or reuses existing logic.
- Add tests covering all branches.

## Scope
**Primary target:** `src/dfastbe/gui/configs.py`

### Refactor
- Introduce a new class (suggested name: `ConfigurationExporter`) that:
  - receives `StateStore.instance()` and builds configuration
  - exposes a `build()` method returning `ConfigParser`
- Replace the `get_configuration` calls by the new class.
- Split into methods with single responsibility, e.g.:
  - `_build_general_section`
  - `_build_detect_section`
  - `_build_erosion_section`
  - `_build_erosion_levels`
  - `_build_filters`
- Consider using `ConfigFile` as a helper/wrapper if it simplifies formatting rules or shared defaults.
- Preserve case-sensitive keys (`config.optionxform = str`).

### Tests
- Add unit tests covering:
  - General section fields and defaults.
  - Detect section (NBank, Linei, DLines list formatting).
  - Erosion section fields, including:
    - Ship params (constant vs variable).
    - `Classes` flag behavior.
    - Filters (`VelFilterDist`, `BedFilterDist`) included only when active.
    - NLevel/RefLevel and optional per-level overrides.
- Add integration-style test that compares full config output against a known-good expected config.
- Unit tests should test one behaviour, all other behaviours should be mocked

### Formatting & Docstrings
- Add class and method docstrings with clear responsibilities.
- Ensure formatting follows out standards [here](https://github.com/Deltares-research/python-tooling/blob/main/docs/python-standards.md).

## Implementation Notes
- If using `ConfigFile`, do so only where it improves clarity, avoid unnecessary re-wrapping.
- Ensure identical output structure and values as current function.

## Definition of Done
-  [ ] `get_configuration` delegates to the new class.
- [ ] The class methods are small, named by responsibility, and documented.
- [ ] Tests cover all branches and pass locally.
- [ ] Output is identical.
- [ ] there are unit tests covering all class methods.

## Out of Scope
- Changing config format.
- UI changes or new features.

## References
- `src/dfastbe/gui/configs.py`

# Time 
- 2 days

### PR DESCRIPTION
**Pull request by riordache12**
# Description

# Issues
- Fixes #182 

## Type of change

Check relevant points.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- Please describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:
- [ ] My branch is updated with the latest changes from the base branch.
- [ ] All pipelines are green?
- [ ] Mkdocs docs is updated with my changes (the examples look as they should in the github pages). 
- [ ] Assigned copilot as a reviewer and addressed the copilot's comments.
- [ ] Updated version number in pyproject.toml.
- [ ] Updated the lock file.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Documentation (docstring/existing mkdocs files) are updated.

### CODE REVIEW

**Comment by MAfarrag** [comment]
@riordache12 can you trigger the copilot reviewer here ?

**Comment by MAfarrag** [comment]
@riordache12 the issue is not written in the pull request

File: src/dfastbe/gui/configs.py [hunk 1]
```diff
@@ -179,6 +180,209 @@ def load_configuration(config_path: Path) -> None:
...
+            # Optional reed parameter
+            if self.state[istr + "_bankReedType"].currentText() != "Use Default":
+                section["Reed" + istr] = self.state[istr + "_bankReedEdit"].text()
+
+            # Optional erosion volume file
+            if self.state[istr + "_eroVolEdit"].text() != "":
+                section["EroVol" + istr] = self.state[istr + "_eroVolEdit"].text()
+
+
 def get_configuration() -> ConfigParser:
```
Comment (MAfarrag) [inline src/dfastbe/gui/configs.py:565]: the old function `get_configuration` is not needed anymore, remove it and use the new class `ConfigurationExporter` directly where the function was called

**Review by MAfarrag — CHANGES_REQUESTED** [review]

File: src/dfastbe/gui/configs.py [hunk 2]
```diff
@@ -179,6 +180,209 @@ def load_configuration(config_path: Path) -> None:
...
+        Args:
+            section: The ConfigParser section to populate.
+        """
+        nlevel = self.state["discharges"].topLevelItemCount()
+        section["NLevel"] = str(nlevel)
+        section["RefLevel"] = self.state["refLevel"].text()
+
+        for i in range(nlevel):
+            istr = str(i + 1)
+            section["SimFile" + istr] = (
```
Comment (MAfarrag) [inline src/dfastbe/gui/configs.py]: can you use fstring instead of these `"any text" + "any text"` ?

**Comment by MAfarrag** [comment]
```
refactor(configs): extract get_configuration into ConfigurationExporter

  Encapsulate the procedural body of get_configuration() in a new
  ConfigurationExporter class with one builder per [General], [Detect]
  and [Erosion] section, plus four sub-builders for ship, bank-strength,
  filter, and per-level discharge parameters. The public function is
  preserved as a thin delegate.

  - Add unit tests in tests/test_gui/test_get_configuration.py covering
    every section builder, the public build() API end-to-end, and the
    nbank == 0 / nlevel == 0 edge cases.
  - Fix DLines empty-case serialization: with NBank == 0 the original
    arithmetic produced a malformed " ]"; the new code emits "[ ]".
    Outputs are byte-identical for NBank >= 1.
  - Wrap BankType (Bank Type + Constant) and per-level ShipType<n>
    (Constant) in str(...) before assignment to ConfigParser; master
    raised TypeError on these branches.
  - Annotate sub-builders with section: SectionProxy.
  - Replace string concatenation with f-strings in the detect and
    erosion-level builders.
  - Cache per-level shipTypeType.currentText() to remove a duplicate
    Qt accessor call in the loop body.
  - Unify file docstrings on Google Args:/Returns: style.
```

File: src/dfastbe/gui/configs.py [hunk 3]
```diff
@@ -179,6 +180,209 @@ def load_configuration(config_path: Path) -> None:
...
+        nbank = self.state["searchLines"].topLevelItemCount()
+        section["NBank"] = str(nbank)
+
+        dlines = "[ "
+        for i in range(nbank):
+            istr = str(i + 1)
+            section["Line" + istr] = self.state["searchLines"].topLevelItem(i).text(1)
+            dlines += self.state["searchLines"].topLevelItem(i).text(2) + ", "
+        dlines = dlines[:-2] + " ]"
+        section["DLines"] = dlines
```
[the remark this answers is not in the document]
Comment (MAfarrag) [reply src/dfastbe/gui/configs.py]: solved

File: tests/test_gui/test_get_configuration.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        section = exporter.config["Erosion"]
+        assert section["NLevel"] == "2"
+        assert section["RefLevel"] == "3"
+        assert section["SimFile1"] == "sim1.nc"
+        assert section["PDischarge1"] == "0.1"
+        assert section["SimFile2"] == "sim2.nc"
+        assert section["PDischarge2"] == "0.2"
+        assert section["ShipType1"] == "1"  # shipTypeSelect index + 1
+        assert section["VShip2"] == "9.9"
+        assert section["EroVol2"] == "vol2.evo"
```
[the remark this answers is not in the document]
Comment (MAfarrag) [reply tests/test_gui/test_get_configuration.py]: solved

File: src/dfastbe/gui/configs.py [hunk 5]
```diff
@@ -179,6 +180,209 @@ def load_configuration(config_path: Path) -> None:
...
+        # Bank strength parameters
+        self._build_bank_strength_parameters(section)
+
+        # Filter parameters
+        self._build_filters(section)
+
+        # Discharge levels
+        self._build_erosion_levels(section)
+
+    def _build_ship_parameters(self, section) -> None:
```
[the remark this answers is not in the document]
Comment (MAfarrag) [reply src/dfastbe/gui/configs.py]: solved

File: src/dfastbe/gui/configs.py (hunk 2 above)
Comment (MAfarrag) [reply src/dfastbe/gui/configs.py]: solved

File: tests/test_gui/test_get_configuration.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
+import pytest
+from unittest.mock import MagicMock
```
[the remark this answers is not in the document]
Comment (MAfarrag) [reply tests/test_gui/test_get_configuration.py]: solved

**Review by MAfarrag — CHANGES_REQUESTED** [review]

File: src/dfastbe/gui/configs.py (hunk 1 above)
Comment (MAfarrag) [reply src/dfastbe/gui/configs.py:565]: keep it as is
